website/content/docs/lifecycle/scope.mdx

```diff
@@ -0,0 +1,200 @@
+
+## Workspace Scoping
+
+We can use [workspaces](/docs/workspaces) as one way to model environments
+such as staging, production, etc. And we might use multiple Kubernetes
+namespaces as a way to separate these stages. The example below uses
+the "default" namespace by default, but uses the "production" namespace
+if the workspace is also "production":
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {}
+
+    workspace "production" {
+      use "kubernetes" {
+        namespace = "production"
+      }
+    }
+  }
+}
+```
+
+Note that we could go further and completely change the plugin in use as well.
+Perhaps we are testing a migration to [Nomad](https://www.nomadproject.io/)
+if we're in the "new-platform" workspace:
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {}
+
+    workspace "production" {
+      use "kubernetes" {
+        namespace = "production"
+      }
+    }
+
+    workspace "new-platform" {
+      use "nomad" {
+        // nomad configuration here
+      }
+    }
+  }
+}
+```
+
+## Label Scoping
+
+The `label` stanza can be used to configure operations based on
+[label selectors](/docs/lifecycle/labels#label-selector-syntax).
+
+~> **Warning!** This is an advanced use case. Most cases can be satisfied
+using the `workspace` stanza which is much simpler to understand and debug.
+Under the covers, the `workspace` stanza is functionally equivalent to a
+label selector of `waypoint/workspace == <name>`.
+
+For example, perhaps we have a different configuration
+depending on if the `region` label is set for operations. (This label
+must be manually set. See [labels](/docs/lifecycle/labels).)
+
+```hcl
+app "my-app" {
+  deploy {
+    use "kubernetes" {}
+
+    label "region == us-east" {
+      use "kubernetes" {
+        // us-east configs
+      }
+    }
+
+    label "region == us-west" {
+      // In us-west we're still migrating from bare ec2
+      use "aws-ec2" {}
+    }
+  }
+}
+```
+
+-> **Mixing:** `label` and `workspace` stanzas can be used together.
+See [multiple match ordering](#multiple-match-ordering) to determine
+the behavior when there are multiple matches.
+
+## Scoped Registry Configuration
+
+The `registry` configuration can be scoped as well, but only within
+the root `build` stanza. This does not limit the functionality of scoping
+registry configuration at all.
+
+For example, the following specifies a custom build and registry configuration
+for the "production" workspace. Note that we don't use `registry` within
+the workspace-scoped build configuration. We still use `registry` within
+the top-level `build` and apply a duplicate `workspace` stanza.
+
+```hcl
+app "my-app" {
+  build {
+    use "docker" {}
+
+    registry {
+      use "docker" {
+        local = true
+      }
+
+      workspace "production" {
+        use "docker" {}
+      }
+    }
+
+    // Alternate build configuration for production.
+    workspace "production" {
```

Comment:
So in the case of `-workspace=production`, this would run a `pack` build and push to a `docker` public registry?

Reply:
That's right!